tasks/Module 3/task_2.py

```diff
@@ -0,0 +1,32 @@
+"""My split function."""
+
+
+def my_split(source_string, separator=None):
+    """My implementation of split() function."""
+    if separator is None:
+        separator = ' '
+    result = []
+    if len(source_string) < len(separator):
+        result.append(source_string)
+        print(result)
+        return result
+    while len(source_string) >= 0:
+        if source_string == separator:
+            result.extend(['', ''])
+            break
+        i = source_string.find(separator)
+        if i == -1:
+            result.append(source_string)
+            print(result)
+            return result
+        result.append(source_string[:i])
+        source_string = source_string[i + len(separator):]
+    print(result)
+    return result
+
+
+if __name__ == '__main__':
+    my_split("Python is cool")
```

Comment:
Do you mind adding tests for this function? I understand the print statements are to verify the result. Tests written with `pytest` can be easily run on the CI/CD level, this way we can ensure that our code is working properly before each deploy.